contracts/src/kleros-v1/KlerosV1Governor.sol

```diff
@@ -0,0 +1,190 @@
+     *  @param _klerosLiquid The trusted arbitrator to resolve potential disputes.
+     *  @param _governor The trusted governor of the contract.
+     *  @param _foreignGateway The trusted gateway that acts as an arbitrator, relaying disputes to v2.
+     */
+    constructor(
+        IKlerosLiquid _klerosLiquid,
+        address _governor,
+        IArbitrator _foreignGateway
+    ) {
+        klerosLiquid = _klerosLiquid;
+        governor = _governor;
+        foreignGateway = _foreignGateway;
+    }
+
+    /** @dev Lets the governor call anything on behalf of the contract.
+     *  @param _destination The destination of the call.
+     *  @param _amount The value sent with the call.
+     *  @param _data The data sent with the call.
+     */
+    function executeGovernorProposal(
+        address _destination,
+        uint256 _amount,
+        bytes calldata _data
+    ) external onlyByGovernor {
+        (bool success, ) = _destination.call{value: _amount}(_data); // solium-disable-line security/no-call-value
+        require(success, "Call execution failed.");
+    }
+
+    /** @dev Changes the `governor` storage variable.
+     *  @param _governor The new value for the `governor` storage variable.
+     */
+    function changeGovernor(address _governor) external onlyByGovernor {
+        governor = _governor;
+    }
+
+    /** @dev Relays disputes from KlerosLiquid to Kleros v2. Only disputes in the evidence period of the initial round can be realyed.
+     *  @param _disputeID The ID of the dispute as defined in KlerosLiquid.
+     */
+    function relayDispute(uint256 _disputeID) external {
+        require(klerosLiquidDisputeIDtoGatewayDisputeID[_disputeID] == 0, "Dispute already relayed");
+        IKlerosLiquid.Dispute memory KlerosLiquidDispute = klerosLiquid.disputes(_disputeID);
+        (uint256[] memory votesLengths, , uint256[] memory totalFeesForJurors, , , ) = klerosLiquid.getDispute(
+            _disputeID
+        );
+
+        // Check that no juror was yet drawn. Add a function to finalize the juror drawing and move the dispute to vote.
+        require(KlerosLiquidDispute.period == IKlerosLiquid.Period.evidence, "Invalid dispute period.");
+        require(votesLengths.length == 1, "Cannot relay appeals.");
+
+        klerosLiquid.executeGovernorProposal(address(this), totalFeesForJurors[0], "");
+
+        uint256 minJurors = votesLengths[0];
+        bytes memory extraData = abi.encode(KlerosLiquidDispute.subcourtID, minJurors);
+        uint256 arbitrationCost = foreignGateway.arbitrationCost(extraData);
+        require(totalFeesForJurors[0] >= arbitrationCost, "Fees not high enough."); // If this doesn't hold at some point, it could be a big issue.
+        uint256 gatewayDisputeID = foreignGateway.createDispute(KlerosLiquidDispute.numberOfChoices, extraData);
+        klerosLiquidDisputeIDtoGatewayDisputeID[_disputeID] = gatewayDisputeID;
+        require(gatewayDisputeID != 0, "ID must be greater than 0.");
```

Comment:
See the above comment about disputeID

Reply:
@shalzz how about having createDispute() return disputeHash instead? There is no requirement for the disputeID to be sequential and the disputeHash won't ever be exposed to the legacy arbitrables or arbitrator.

Reply:
Maybe you forgot, the gateway contracts have to follow the arbitrable/arbitator interfaces. since the KlerosV1 proxy will not be the only user of the gateway contracts.

Reply:
This is not a new idea, it has been proposed by Henrique already a few months ago.

The disputeHash is a bytes32 so obviously it fits in a uint256 which satisfies the Arbitrator interface. As I said it won't be exposed to the v1 contracts. For future v2 integrations talking to the gateways, it will be known in advance not to expect sequential IDs which again is not a requirement from the standard.

Reply:
The disputeHash requires a disputeID as part of the pre-image. Do you mean to remove the disputeID from the pre-image and use the disputeHash as the  disputeID or just overwrite the disputeID with the generated disputeHash?

Reply:
Okay no worries let's keep this part as it, now that the gatewayDisputeId starts from 1. 
a9df0ab